application-workflow-detailed.md

```diff
@@ -0,0 +1,442 @@
+
+Currently in local storage:
+```
+OPENID_CONFIGURATION
+RP_PRIVATE_KEY
+RP_PUBLIC_KEY
+```
+
+#### 7. Requests OP JWKs
+
+Now that the RP's Public/Private keys are generated. The RP needs to be aware of the OP's public key. To do so a request should be made to the `jwks_uri` from the openid-configuration:
+
+```
+GET https://secureauth.example/jwks
+```
+
+#### 8. Returns OP JWKs
+
+The [JSON Web Key Set (JWKS)](https://auth0.com/docs/jwks) is returned. These will eventually be used to validate the signature of the token the OP will eventually issue. It is recommended that the result of this request should be saved to local storage.
+
+Response body:
+```json
+{  
+  "keys":[  
+    {  
+      "kty":"RSA",
+      "kid":"xeOjes9u3AcU4LBzcanEM7pZLwSlxaN7U62ZzOBDQuw",
+      "alg":"RS256",
+      "key_ops":[  
+        "verify"
+      ],
+      "e":"AQAB",
+      "n":"oB2LgkiZZ5iLAz1d4ua7sVxdbzY2nIRkDtf4UE08mWsD6UYRzLR98_gMAfnKB8i9yPCQkxfA5w_SZq6Y7odG1qSwLHM2mb_O2GSvY9kaG00UpeeEJCR19c7Jkcmq3GXh4yujnm2TFQ6YAzYNgrXkHlusaFUApJaQN6zr4AvmR_vX_5i__Ku7nuU-GbaV75LSr8o0QANdYFF0ooz5DJvydPplF8mO9_oD7ceSNLWP1AXlFs5JH6MEhH02dELb4-zeLcVzhoqON60cABTpbYSf1lLbYZsVUQ3cYE9CxXaByY2YNuQgc0k29mSmUvwEs0hNA5xUcE3-y_qKpYKniErb9Q"
+    }
+  ]
+}
+```
+
+Currently in local storage:
+```
+OPENID_CONFIGURATION
+RP_PRIVATE_KEY
+RP_PUBLIC_KEY
+OP_JWKS
+```
+
+#### 9. Sends Dynamic Registration Parameters
+
+Now we have everything we need to perform [dynamic client registration](https://openid.net/specs/openid-connect-registration-1_0.html). Because each Solid user could have a different OP, it is not feasible to expect the developers of RPs to manually register with every OP. Therefore, the client (RP) is registered dynamically. A request is sent to the OP's `registration_endpoint`:
+
+```
+POST https://secureauth.example/register
+```
+Data:
+```json
+{
+  "grant_types": ["implicit"],
+  "issuer": "https://secureauth.example",
+  "redirect_uris": ["https://www.decentphotos.example/"],
+  "response_types": ["id_token token"],
+  "scope": "openid profile"
+}
+```
+
+Each of these communicates something about the new client to the OP:
+ - `grant_types`: A list of [OIDC grant types](http://docs.identityserver.io/en/latest/topics/grant_types.html) this client will use. `implicit` is great for web applications.
+ - `issuer`: Alice's OP
+ - `redirect_uris`: Redirect uris provided at the client registration stage state which redirect uris are valid during the authorization stage.
+ - `response_types`: A list of [OIDC response types](https://openid.net/specs/oauth-v2-multiple-response-types-1_0.html) the client can use. `id_token token` means that it should return both an Id Token (needed to identify this application later) and an access token.
+ - `scope`: OIDC uses scope as a way of defining what a client can have acces to. However, Solid has it's own access control system, so scope will always be `openid profile`
```

Comment:
typo: can have acces to -> can have access to
typo: Solid has it's -> Solid has its